cdp_scrapers/legistar_utils.py

```diff
@@ -395,6 +395,48 @@ class Status(enum.IntEnum):
     uris: Optional[List[ContentURIs]] = None
 
 
+def parse_video_page_url(video_page_url: str, client: str):
+    """
+    Return URLs for videos and captions from a Legistar/Granicus-hosted video web page
+
+    Parameters
+    ----------
+    video_page_url: str
+        The URL for the page of the legistar video
+    client: str
+        Which legistar client to target. Ex: "seattle"
+
+    Returns
+    -------
+    ContentUriScrapeResult
+        uris: Optional[List[ContentURIs]]
+            URIs for video and optional caption
+    """
+    try:
+        with urlopen(video_page_url) as resp:
+            # now load the page to get the actual video url
+            soup = BeautifulSoup(resp.read(), "html.parser")
+
+            if client in video_page_parser:
+                # we already know which format parser to call
+                uris = video_page_parser[client](client, soup)
+            else:
+                for parser in all_parsers:
+                    uris = parser(client, soup)
+                    if uris is not None:
+                        # remember so we just call this from here on
+                        video_page_parser[client] = parser
+                        log.debug(f"{parser} for {client}")
+                        break
+                else:
+                    uris = None
+    except HTTPError as e:
+        log.debug(f"Error opening {video_page_url}:\n{str(e)}")
+        return (ContentUriScrapeResult.Status.ResourceAccessError, None)
```

Comment:
This isn't quite right Shak. `get_legistar_content_uris()` needs to return this in case of `HTTPError`, not here in `parse_video_page_url()`. So I would move the `try...except` back to `get_legistar_content_uris()`, where you call this function.